sshfs-win.c

```diff
@@ -314,6 +321,16 @@ static int do_svc(int argc, char *argv[])
         sshfs, SSHFS_ARGS, idmap, authmeth, volpfx, portopt, remote, argv[2], 0,
     };
 
+    if (strlen(portopt) == 0)
+    {
+        int portopt_idx = 10;
```

Comment:
This and the loop after it is quite subtle in what it does. For example, it is not immediately obvious why `portopt_idx` is `10`. (When I was looking at the `sshfs_argv` array I could have sworn that `portopt_idx` should be `5` because I did not immediately realize that `SSHFS_ARGS` is an array.)

This also makes it quite fragile. For example, what if someone changes `SSHFS_ARGS` to add or remove elements, but never updates the `portopt_idx` starting value?

My suggestion would be to compute `portopt_idx` by looking inside `sshfs_argv` for `portopt` using another loop before the current one. Something like (untested):

```C
        /* if are not passing a port option remove it from sshfs_argv */
        int portopt_idx = 0;
        while (sshfs_argv[portopt_idx] != portopt)
            portopt_idx++;
```

Reply:
Thanks, fixed.